web/src/settings_invites.js

```diff
@@ -214,7 +263,85 @@ export function on_load_success(invites_data, initialize_event_handlers) {
         $(".dialog_submit_button").attr("data-is-multiuse", meta.is_multiuse);
     });
 
-    $(".admin_invites_table").on("click", ".resend", (e) => {
+    $(".admin_invites_table").on("click", ".edit-invite", (e) => {
+        // This click event must not get propagated to parent container otherwise the modal
+        // will not show up because of a call to `close_active_modal` in `settings.js`.
+        e.stopPropagation();
+        e.preventDefault();
+
+        const invite_id = Number.parseInt($(e.currentTarget).attr("data-invite-id"), 10);
+        const invite = meta.invites.find((invite) => invite.id === invite_id && invite.is_multiuse);
+
+        const html_body = render_edit_invite_user_modal({
+            invite_as_options: settings_config.user_role_values,
+            is_admin: page_params.is_admin,
+            is_owner: page_params.is_owner,
+            streams: get_invite_streams(),
+            notifications_stream: stream_data.get_notifications_stream(),
+            invite,
+        });
+
+        function invite_user_modal_post_render() {
+            $("#invite-user-modal .dialog_submit_button").prop("disabled", true);
+            for (const stream_id of invite.streams) {
+                $(`[value=${stream_id}]`).prop("checked", true);
+            }
+            $("#invite_as").val(invite.invited_as);
+
+            function enable_invite_submit_button() {
+                $("#invite-user-modal .dialog_submit_button").prop("disabled", false);
+            }
+            function disable_invite_submit_button() {
+                $("#invite-user-modal .dialog_submit_button").prop("disabled", true);
+            }
+            function state_unchanged() {
+                const initial_streams = invite.streams.sort();
+                let selected_streams = [];
+                $("#streams_to_add input:checked").each((_, stream) => {
+                    selected_streams.push(Number.parseInt(stream.value, 10));
+                });
+                selected_streams = selected_streams.sort();
+                return (
+                    selected_streams.length === initial_streams.length &&
+                    selected_streams.every((val, index) => val === initial_streams[index]) &&
+                    Number.parseInt($("#invite_as").val(), 10) === invite.invited_as
+                );
+            }
+
+            // Checks the state of the modal and enables or disables the save changes button
+            function toggle_submit_button() {
```

Comment:
Adding this function is probably overkill. You can just call `$("#invite-user-modal .dialog_submit_button").prop("disabled", state_changed());`.